File: pyg_01/pyg_manager_web_01/src/main/webapp/js/controller/indexController.js
app.controller('indexController',function ($scope,$http,indexService,$controller) {
    $controller('baseController', {$scope:$scope});//继承baseController,将父模块中的$scope赋值给子模块的$scope
    $scope.username = '';//当前登录的用户名
    $scope.findLoginUser=function(){
        indexService.findLoginUser().success(function (res) {
            $scope.username= res.username;
        })
    };

});

File: pyg_01/pgy_shop_web_01/src/main/java/com/pinyougou/shop/controller/indexController.java
package com.pinyougou.shop.controller;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/index")
public class indexController {

    @RequestMapping("/findLoginUser")
    public Map<String,String> findLoginUser(){
        //从springsecurity中获取用户信息
        String name = SecurityContextHolder.getContext().getAuthentication().getName();

        //将用户封装到key为username的map中.返回
        Map map=new HashMap<>();
        map.put("username", name);
        return map;
    }

}
